README.md

```diff
@@ -95,35 +95,65 @@ These operations add to or update data in Immutable's databases and typically re
-
 1. [Generate your own by obtaining and using the user's private keys](#1-generate-your-own-signers)
 2. [Use our Wallet SDK to connect to a user's wallet application](#2-generate-signers-using-the-wallet-sdk)
 
 The first option, where an application obtains a user's private key directly, is risky because these keys allow anyone in possession of them full control of an account.
 
 The second option provides an application with an interface to the user's account by prompting the user to connect with their wallet application (ie. mobile or browser wallet). Once connected the app can begin asking the user to sign transactions and messages without having to reveal their private key.
 
-As Immutable X enables applications to execute signed transactions on both Ethereum (layer 1) and StarkEx (layer 2), signers are required for both these layers.
+### Pre-requisite: User must have Ethereum (L1) and Stark (L2) keys
 
-### 1. Generate your own signers
+As Immutable X enables applications to execute signed transactions on both Ethereum (layer 1) and StarkEx (layer 2), signers are required for both these layers. In order to generate an Ethereum or Stark signer, a user's Ethereum or Stark private key is required.
+
+The Core SDK can be used to generate a Stark key for a user. Previously, this key was deterministically generated so that the same key could always be obtained using the user's Ethereum key. This meant that if someone had access to a user's L1 (Ethereum) private keys, they could also obtain the user's L2 (Stark) private keys. This was [changed in v1.0.0-beta3](https://docs.x.immutable.com/sdk-docs/core-sdk-ts/core-sdk-migration-guide/#what-has-changed-in-core-sdk-v100-beta3) of the Core SDK, so that Stark keys are now randomly generated and non-reproducible.
+
+If your user has a Stark key that was generated using the previous, deterministic method, you can still retrieve their Stark private key by using the [generateLegacyStarkPrivateKey()](https://github.com/immutable/imx-core-sdk/blob/83f800956f541f338b3267ec7cb16e039182dfa6/src/utils/stark/starkCurve.ts#L152) method:
+```ts
+import { AlchemyProvider } from '@ethersproject/providers';
+import { Wallet } from '@ethersproject/wallet';
+import { generateLegacyStarkPrivateKey } from '@imtbl/core-sdk';
+
+// Create Ethereum signer
+const ethNetwork = 'goerli'; // Or 'mainnet'
+const provider = new AlchemyProvider(ethNetwork, YOUR_ALCHEMY_API_KEY);
+const ethSigner = new Wallet(YOUR_PRIVATE_ETH_KEY).connect(provider);
+
+// Get the legacy Stark private key
+const starkPrivateKey = generateLegacyStarkPrivateKey(ethSigner);
+```
 
-The Core SDK provides functionality for applications to generate Stark (L2) [private keys](/src/utils/stark/starkCurve.ts#L108) and [signers](/src/utils/stark/starkSigner.ts#L60).
+If they do not yet have a Stark key, you can use the Core SDK to generate one for them using [generateStarkPrivateKey()](/src/utils/stark/starkCurve.ts#L108):
```

Comment:
I don't think we can say this line. 

This recommendation only applies to wallets that will be used to mint assets. Could we avoid using the term "the user" or "they" and instead address the content originators in first person?

Reply:
We should provide a clear distinction of when to use `generateLegacyStarkPrivateKey` and when to use `generateStarkPrivateKey` and list out the pros and cons of each path so that the SDK users are better educated on which path to choose.

Reply:
I've added a table to outline this @suluvai